client/service/src/builder.rs

```diff
@@ -329,611 +296,107 @@ pub fn new_client<E, Block, RA>(
-	) -> Result<ServiceBuilder<TBl, TRtApi, TCl, TFchr, TSc, TImpQu, TFprb, TFpp,
-		TExPool, TRpc, Backend>, Error>
-	where TSc: Clone, TFchr: Clone {
-		Ok(ServiceBuilder {
-			informant_prefix: informant_prefix,
-			..self
-		})
-	}
+		),
+	);
+	let fetcher = Arc::new(sc_network::config::OnDemand::new(fetch_checker));
+	let backend = crate::client::light::new_light_backend(light_blockchain);
+	let remote_blockchain = backend.remote_blockchain();
+	let client = crate::client::light::new_light(
+		backend.clone(),
+		config.chain_spec.as_storage_builder(),
+		executor,
+		Box::new(task_manager.spawn_handle()),
+		config.prometheus_config.as_ref().map(|config| config.registry.clone()),
+	)?;
+
+	Ok(((client, backend, keystore, task_manager), fetcher, remote_blockchain))
 }
 
-/// Implemented on `ServiceBuilder`. Allows running block commands, such as import/export/validate
-/// components to the builder.
-pub trait ServiceBuilderCommand {
-	/// Block type this API operates on.
-	type Block: BlockT;
-	/// Native execution dispatch required by some commands.
-	type NativeDispatch: NativeExecutionDispatch + 'static;
-	/// Starts the process of importing blocks.
-	fn import_blocks(
-		self,
-		input: impl Read + Seek + Send + 'static,
-		force: bool,
-		binary: bool,
-	) -> Pin<Box<dyn Future<Output = Result<(), Error>> + Send>>;
-
-	/// Performs the blocks export.
-	fn export_blocks(
-		self,
-		output: impl Write + 'static,
-		from: NumberFor<Self::Block>,
-		to: Option<NumberFor<Self::Block>>,
-		binary: bool
-	) -> Pin<Box<dyn Future<Output = Result<(), Error>>>>;
-
-	/// Performs a revert of `blocks` blocks.
-	fn revert_chain(
-		&self,
-		blocks: NumberFor<Self::Block>
-	) -> Result<(), Error>;
-
-	/// Re-validate known block.
-	fn check_block(
-		self,
-		block: BlockId<Self::Block>
-	) -> Pin<Box<dyn Future<Output = Result<(), Error>> + Send>>;
-
-	/// Export the raw state at the given `block`. If `block` is `None`, the
-	/// best block will be used.
-	fn export_raw_state(
-		&self,
-		block: Option<BlockId<Self::Block>>,
-	) -> Result<Storage, Error>;
+pub struct ServiceDescriptor<TBl: BlockT, TRtApi, TSc, TImpQu, TExPool, TBackend, TExec, TRpcBuilder> {
+	pub config: Configuration,
+	pub client: Arc<Client<TBackend, TExec, TBl, TRtApi>,>,
+	pub backend: Arc<TBackend>,
+	pub task_manager: TaskManager,
+	pub keystore: Arc<RwLock<Keystore>>,
+	pub on_demand: Option<Arc<OnDemand<TBl>>>,
+	pub select_chain: Option<TSc>,
+	pub import_queue: TImpQu,
+	pub finality_proof_request_builder: Option<BoxFinalityProofRequestBuilder<TBl>>,
+	pub finality_proof_provider: Option<Arc<dyn FinalityProofProvider<TBl>>>,
+	pub transaction_pool: Arc<TExPool>,
+	pub remote_blockchain: Option<Arc<dyn RemoteBlockchain<TBl>>>,
+	pub background_tasks: Vec<(&'static str, BackgroundTask)>,
+	pub block_announce_validator_builder: Option<Box<dyn FnOnce(Arc<Client<TBackend, TExec, TBl, TRtApi>,>) -> Box<dyn BlockAnnounceValidator<TBl> + Send> + Send + 'static>>,
```

Comment:
Not sure why you kept that one as a builder :grin: Oh I get it, it was added later. You can remove the entire FnOnce thingy and keep only `Option<Box<dyn BlockAnnounceValidator<TBl>>` here I think.